test/unittest/blas1/blas1_axpy_test.cpp

```diff
@@ -194,6 +190,188 @@ TYPED_TEST(BLAS_Test, axpy_test_vpr) {
+  DEBUG_PRINT(std::cout << "strd == " << strd << std::endl);
+  // setting alpha to some value
+  ScalarT alpha(1.54);
+  // creating three vectors: vX, vY and vZ.
+  // the for loop will compute axpy for vX, vY
+  std::vector<ScalarT> vX(size);
+  std::vector<ScalarT> vY(size);
+  std::vector<ScalarT> vZ(size, 0);
+  TestClass::set_rand(vX, size);
+  TestClass::set_rand(vY, size);
+
+  // compute axpy in a for loop and put the result into vZ
+  for (size_t i = 0; i < size; ++i) {
+    if (i % strd == 0) {
+      vZ[i] = alpha * vX[i] + vY[i];
+    } else {
+      vZ[i] = vY[i];
+    }
+  }
+
+  SYCL_DEVICE_SELECTOR d;
+  auto q = TestClass::make_queue(d);
+  Executor<ExecutorType> ex(q);
+  auto gpu_vX = blas::helper::make_sycl_iterator_buffer<ScalarT>(vX, size);
+  auto gpu_vY = blas::helper::make_sycl_iterator_buffer<ScalarT>(vY, size);
+  _axpy<8>(ex, (size + strd - 1) / strd, alpha, gpu_vX, strd, gpu_vY, strd);
+  auto event = ex.copy_to_host(gpu_vY, vY.data(), size);
+  ex.wait(event);
+
+  // check that both results are the same
+  for (size_t i = 0; i < size; ++i) {
+    ASSERT_NEAR(vZ[i], vY[i], prec);
+  }
+}
+
+TYPED_TEST_CASE(BLAS_Test, BlasTypes);
+REGISTER_SIZE(256, DISABLED_axpy_test_tiled)
+REGISTER_STRD(1, DISABLED_axpy_test_tiled)
+REGISTER_PREC(float, 1e-4, DISABLED_axpy_test_tiled)
+REGISTER_PREC(double, 1e-6, DISABLED_axpy_test_tiled)
+REGISTER_PREC(std::complex<float>, 1e-4, DISABLED_axpy_test_tiled)
+REGISTER_PREC(std::complex<double>, 1e-6, DISABLED_axpy_test_tiled)
+
+TYPED_TEST(BLAS_Test, DISABLED_axpy_test_tiled) {
+  using ScalarT = typename TypeParam::scalar_t;
+  using ExecutorType = typename TypeParam::executor_t;
+  using TestClass = BLAS_Test<TypeParam>;
+  using test = class DISABLED_axpy_test_tiled;
+
+  size_t size = TestClass::template test_size<test>();
+  long strd = TestClass::template test_strd<test>();
+  ScalarT prec = TestClass::template test_prec<test>();
+
+  DEBUG_PRINT(std::cout << "size == " << size << std::endl);
+  DEBUG_PRINT(std::cout << "strd == " << strd << std::endl);
+  // setting alpha to some value
+  ScalarT alpha(1.54);
+  // creating three vectors: vX, vY and vZ.
+  // the for loop will compute axpy for vX, vY
+  std::vector<ScalarT> vX(size);
+  std::vector<ScalarT> vY(size);
+  std::vector<ScalarT> vZ(size, 0);
+  TestClass::set_rand(vX, size);
+  TestClass::set_rand(vY, size);
+
+  // compute axpy in a for loop and put the result into vZ
+  for (size_t i = 0; i < size; ++i) {
+    if (i % strd == 0) {
+      vZ[i] = alpha * vX[i] + vY[i];
+    } else {
+      vZ[i] = vY[i];
+    }
+  }
+
+  SYCL_DEVICE_SELECTOR d;
+  auto q = TestClass::make_queue(d);
+  Executor<ExecutorType> ex(q);
+  auto gpu_vX = blas::helper::make_sycl_iterator_buffer<ScalarT>(vX, size);
+  auto gpu_vY = ex.template allocate<ScalarT>(size);
+  ex.copy_to_device(vY.data(), gpu_vY, size);
+  _axpy<8>(ex, (size + strd - 1) / strd, alpha, gpu_vX, strd, gpu_vY, strd);
```

Comment:
why 8? is it hardcoding the tile-size? use a clear constant if thats the case

Reply:
8 is a tile size that we know works here. I've pulled it out into a `constexpr size_t const` variable so that readers can see it explicitly.